tests/test_indexing.cc

```diff
@@ -124,7 +125,7 @@ TEST(test_indexing, test_diagonal_set_subset) {
   expected = expected_diag.asDiagonal();
   to = DiagonalMatrixXd(x);
   set_subset(from, idx, &to);
-  EXPECT_EQ(to, expected);
+  EXPECT_TRUE(to == expected);
```

Comment:
Why was this needed? Doesn't `EXPECT_EQ` use `==`?

Reply:
I have no clue why `EXPECT_EQ` doesn't work with `Eigen::DiagonalMatrix<double, Dynamic>`.  It probably has something to do with the lack of `==` operator (I had to define a free standing one).